### PR TITLE
Adds databaseChangelogSQL mdc key (DAT-13092)

### PR DESCRIPTION
**Pull request by abrackx**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description
Adds databaseChangelogSQL mdc key, populates via StandardChangelogHistoryService and StandardLockService. Moves changesetSql mdc population into Changeset class. 

## Things to be aware of
‼️ This makes it so that end delimiters (ex: `;`) are included in the changesetSql mdc value. ‼️ I can control this, but because I'm now handling the sql at a level above statement execution it made the most sense to take this route to me. We're not always, but sometimes working with more than one statement per change.

### CODE REVIEW

File: liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java [hunk 1]
```diff
@@ -1484,4 +1491,14 @@ private void addChangeSetMdcProperties() {
         Scope.getCurrentScope().addMdcValue(MdcKey.CHANGESET_LABEL, labelMdc);
         Scope.getCurrentScope().addMdcValue(MdcKey.CHANGESET_CONTEXT, contextsMdc);
     }
+
+    private void addSqlMdc(Change change, Database database, boolean generateRollbackStatements) throws RollbackImpossibleException {
+        if (change.supports(database)) {
```
Comment (suryaaki2) [inline liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java:1496]: What would happen if change doesn't support MDC? Can you also add a javadoc comment to the method?

**Review by suryaaki2 — CHANGES_REQUESTED** [review]

File: liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java (hunk 1 above)
Comment (abrackx) [reply liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java:1496]: If the change doesn't support the database then we do not add it to the mdc, as we're going to skip it in execution.

**Review by suryaaki2 — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]
Nice, getJdbcExecutor helps to make sure that we are always logging to MDC.

File: liquibase-core/src/main/java/liquibase/util/SqlUtil.java [hunk 2]
```diff
@@ -341,4 +346,22 @@ public static String replacePredicatePlaceholders(Database database, String pred
...
+     * @param sqlGeneratorFactory the SqlGeneratorFactory instance to use to generate the sql
+     * @param database the database to generate sql against
+     * @return the sql string or an empty string if there are no statements to generate
+     */
+    public static String getSqlString(SqlStatement statement, SqlGeneratorFactory sqlGeneratorFactory, Database database) {
+        Sql[] sqlStatements = sqlGeneratorFactory.generateSql(statement, database);
+        if (sqlStatements != null) {
+            return Arrays.stream(sqlStatements)
+                    .map(sql -> sql.toSql() + sql.getEndDelimiter())
+                    .collect(Collectors.joining("\n"));
```
Comment (StevenMassaro) [inline liquibase-core/src/main/java/liquibase/util/SqlUtil.java:362]: Should this use System.lineseparator? Maybe it is best to be consistent with the line ending.

**Review by StevenMassaro — APPROVED** [review]

**Pull request merged**